app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -138,6 +138,15 @@ def upload_package_profile
       render :json => Resources::Candlepin::Consumer.get(@host.subscription_facet.uuid)
     end
 
+    api :PUT, "/consumers/:id/tracer", N_("Update services requiring restart")
+    param :traces, Hash, :required => true
+    def upload_tracer_profile
+      User.as_anonymous_admin do
+        async_task(::Actions::Katello::Host::UploadTracerProfile, @host, params[:traces])
```

Comment:
I imagine these tracer uploads are going to be fairly small?  You could probably get away with doing this outside of a dynflow action by just doing the `host.import_tracer_profile(profile)` within this controller.  It doesn't cause any communication with candlepin or pulp, and I wouldn't expect it to be a long process.